data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/PipelineTransformer.java

```diff
@@ -117,6 +119,8 @@ private void buildPipelineFromConfiguration(
             LOG.info("Building buffer for the pipeline [{}]", pipelineName);
             final Buffer pipelineDefinedBuffer = pluginFactory.loadPlugin(Buffer.class, pipelineConfiguration.getBufferPluginSetting(), source.getDecoder());
 
+            if (pipelineDefinedBuffer.isByteBuffer())
+                acknowledgementsEnabled = true;
```

Comment:
Should this be set to true for all sub-pipelines just because one sub-pipeline uses persistent buffer?

Reply:
Yes. I have added comment and the test case also validates this.